src/MSBuild/XMake.cs

```diff
@@ -1207,8 +1210,9 @@ private static void ResetGatheringSwitchesState()
 
                 ToolsetDefinitionLocations toolsetDefinitionLocations = ToolsetDefinitionLocations.Default;
 
-                bool preprocessOnly = preprocessWriter != null && !FileUtilities.IsSolutionFilename(projectFile);
-                bool targetsOnly = targetsWriter != null && !FileUtilities.IsSolutionFilename(projectFile);
```

Comment:
It is not clear to me what 'Only' means here. The two switches are not exclusive of each other and both can be performed in one run. So 'Only' can't mean that only preprocess or only targets is being performed.

Reply:
> The two switches are not exclusive of each other and both can be performed in one run.

Really? That sounds broken to me. I think they should be mutually exclusive.

Reply:
I recall testing and seeing that the switches were not exclusive of each other but in the current shipping version they are mutually exclusive and whichever comes second on the command line is silently ignored.

The two switches are orthogonal operations to each other. They don't conflict with each other and they don't modify each other. From that perspective they don't need to mutually exclusive.

Reply:
I didn't make this change for the variable names because the two switches are not mutually exclusive.

There would need to be changes to the switch processing logic to make these two switches mutually exclusive.